### PR TITLE
test: fixup and improvements for rbd-mirror test

### PR DESCRIPTION
**Pull request by trociny**
Signed-off-by: Mykola Golub <mgolub@mirantis.com>

### CODE REVIEW

**Comment by trociny** [comment]
To troubleshoot a test failure I use the following commands:

From src dir start the test in NOCLEANUP mode (so it possible to check its state after failure) and with TEMPDIR pointing to known location):
```
RBD_MIRROR_NOCLEANUP=1 RBD_MIRROR_TEMDIR=/tmp/tmp.rbd_mirror ../qa/workunits/rbd/rbd_mirror.sh
```
After failure cd to TEMPDIR and check status:
```
cd /tmp/tmp.rbd_mirror
less rbd-mirror.log
rbd --cluster remote -p mirror journal status --image test
ceph --admin-daemon rbd-mirror.asok help
...
```
Run cleanup:
```
cd - #SRCDIR
RBD_MIRROR_TEMDIR=/tmp/tmp.rbd_mirror ../qa/workunits/rbd/rbd_mirror.sh cleanup
```

**Comment by jdurgin** [comment]
Could you add that comment on how to use it to the script (perhaps a comment on using the script at the top of the file)?

File: qa/workunits/rbd/rbd_mirror.sh [hunk 1]
```diff
@@ -113,26 +118,50 @@ flush()
...
 }
 
+wait_for_image_replay_started()
+{
+    local image=$1
+
+    wait_for_image_replay_state ${image} started
+}
+
+wait_for_image_replay_stopped()
```
Comment (jdurgin) [inline qa/workunits/rbd/rbd_mirror.sh]: looks like this isn't used yet

Comment (trociny) [inline qa/workunits/rbd/rbd_mirror.sh]: Yes, I was going to use this to test live image replayer removal after an image is removed (journaling is disabled). But right now I can't remove the image when it is replayed due to "Device or resource busy" error. Still I think it is desirable feature and we will be able to do this in some future, and this function will be useful then.

Comment (jdurgin) [inline qa/workunits/rbd/rbd_mirror.sh]: makes sense to leave it in then. I think Jason fixed that EBUSY error with his aio_notify patch that's now in master.

**Comment by trociny** [comment]
Rebased after adding "how to use" into the scrip.

**Pull request merged**